core/field_matrix.js

```diff
@@ -23,544 +23,550 @@
+          'rx': nodePad, 'ry': nodePad,
+          'fill': fill
+        }, button);
+      }
     }
+    return button;
   }
-  // Div for lower button menu
-  var buttonDiv = document.createElement('div');
-  // Button to clear matrix
-  var clearButtonDiv = document.createElement('div');
-  clearButtonDiv.className = 'scratchMatrixButtonDiv';
-  var clearButton = this.createButton_(this.sourceBlock_.colourSecondary_);
-  clearButtonDiv.appendChild(clearButton);
-  // Button to fill matrix
-  var fillButtonDiv = document.createElement('div');
-  fillButtonDiv.className = 'scratchMatrixButtonDiv';
-  var fillButton = this.createButton_('#FFFFFF');
-  fillButtonDiv.appendChild(fillButton);
-
-  buttonDiv.appendChild(clearButtonDiv);
-  buttonDiv.appendChild(fillButtonDiv);
-  div.appendChild(buttonDiv);
-
-  Blockly.DropDownDiv.setColour(this.sourceBlock_.getColour(),
-      this.sourceBlock_.getColourTertiary());
-  Blockly.DropDownDiv.setCategory(this.sourceBlock_.getCategory());
-  Blockly.DropDownDiv.showPositionedByBlock(this, this.sourceBlock_);
-
-  this.matrixTouchWrapper_ =
-      Blockly.bindEvent_(this.matrixStage_, 'mousedown', this, this.onMouseDown);
-  this.clearButtonWrapper_ =
-      Blockly.bindEvent_(clearButton, 'click', this, this.clearMatrix_);
-  this.fillButtonWrapper_ =
-    Blockly.bindEvent_(fillButton, 'click', this, this.fillMatrix_);
-
-  // Update the matrix for the current value
-  this.updateMatrix_();
-
-};
-
-this.nodeCallback_ = function(e, num) {
-  console.log(num);
-};
 
-/**
- * Make an svg object that resembles a 3x3 matrix to be used as a button.
- * @param {string} fill The color to fill the matrix nodes.
- * @return {SvgElement} The button svg element.
- */
-Blockly.FieldMatrix.prototype.createButton_ = function(fill) {
-  var button = Blockly.utils.createSvgElement('svg', {
-    'xmlns': 'http://www.w3.org/2000/svg',
-    'xmlns:html': 'http://www.w3.org/1999/xhtml',
-    'xmlns:xlink': 'http://www.w3.org/1999/xlink',
-    'version': '1.1',
-    'height': Blockly.FieldMatrix.MATRIX_NODE_SIZE + 'px',
-    'width': Blockly.FieldMatrix.MATRIX_NODE_SIZE + 'px'
-  });
-  var nodeSize = Blockly.FieldMatrix.MATRIX_NODE_SIZE / 4;
-  var nodePad = Blockly.FieldMatrix.MATRIX_NODE_SIZE / 16;
-  for (var i = 0; i < 3; i++) {
-    for (var n = 0; n < 3; n++) {
-      Blockly.utils.createSvgElement('rect', {
-        'x': ((nodeSize + nodePad) * n) + nodePad,
-        'y': ((nodeSize + nodePad) * i) + nodePad,
-        'width': nodeSize, 'height': nodeSize,
-        'rx': nodePad, 'ry': nodePad,
-        'fill': fill
-      }, button);
+  /**
+   * Redraw the matrix with the current value.
+   * @private
+   */
+  updateMatrix_() {
+    for (var i = 0; i < this.matrix_.length; i++) {
+      if (this.matrix_[i] === '0') {
+        this.fillMatrixNode_(this.ledButtons_, i, this.sourceBlock_.getColourSecondary());
+        this.fillMatrixNode_(this.ledThumbNodes_, i, this.sourceBlock_.getColour());
```

Comment:
This actually looks really bad in high contrast mode. Not sure if/where we should track accessibility things.

![image](https://github.com/gonfunko/scratch-blocks/assets/25440652/f50c70c5-4754-4980-ac15-6b241c80396a)

Reply:
Probably just file an issue?

Reply:
Against us or against them though? like are we going to fix these things?

Reply:
I think against us, at least for now. We can triage and discuss how they actually get resolved when we're to that point.